packaging/rpm/README.org

```diff
@@ -11,33 +11,67 @@ commit made in (1.) on top of this and start over from the beginning. This
 fix minor errors.
 
-** Prerequisites
-
-This assumes that the Fedora and RHEL dist-git repositories are checked out in
-=$FEDORA_DISTGIT_DIR= and =$RHEL_DISTGIT_DIR= respectively, and that the user
-has valid Kerberos tickets exist for Fedora and RHEL infrastructure (=kinit
-username@FEDORAPROJECT.ORG= and =kinit username@REDHAT.COM respectively=). The
-dist-git repositories can be found here:
-
-Fedora: https://src.fedoraproject.org/rpms/xdp-tools (master branch)
-RHEL: http://pkgs.devel.redhat.com/cgit/rpms/xdp-tools/ (rhel-8.3.0 branch)
 
 ** Steps
 
 To release a new version of xdp-tools, follow these steps:
 
-1. Bump version in =version.mk= and =packacing/rpm/xdp-tools.spec= -- don't
+1. Make sure you have valid Kerberos tickets exist for the Fedora and RHEL
+   infrastructure:
+
+   #+begin_src sh
+   kinit <username>@FEDORAPROJECT.ORG
+   kinit <username>@REDHAT.COM
+   #+end_src
+
+2. Checkout this git repository as well as the RHEL and Fedora dist-git's.
+   Also, make sure you move to the correct branch for RHEL and Fedora.
+   Below is an example of how you could do this:
+
+   #+begin_src sh
+   export VERSION="<version, for example 1.0.0~beta3>"
+   mkdir release_$VERSION
+   cd release_$VERSION
+   git clone git@github.com:xdp-project/xdp-tools.git
+   cd xdp-tools
+   export XDP_TOOLS_DIR=$(pwd)
+
+   rhpkg clone xdp-tools xdp-tools-RHEL
+   cd xdp-tools-RHEL
+   rhpkg switch-branch -l
+   rhpkg switch-branch rhel-8.3.0
+   export RHEL_DISTGIT_DIR=$(pwd)
+   cd ..
+
+   fedpkg clone xdp-tools xdp-tools-FEDORA
+   cd xdp-tools-FEDORA
+   fedpkg switch-branch -l
+   export FEDORA_DISTGIT_DIR=$(pwd)
+   cd ..
+   #+end_src
+
+3. Bump version in =version.mk= and =packacing/rpm/xdp-tools.spec= -- don't
    forget a changelog entry in the latter. Commit this, bug don't tag and push
    until the rest of the steps below completed successfully.
 
-2. Run =make archive= to generate a source tarball (xdp-tools-$VERSION.tar.gz).
+4. Run =make archive= to generate a source tarball (xdp-tools-$VERSION.tar.gz).
+   #+begin_src sh
+   cd $XDP_TOOLS_DIR
+   git submodule init
+   git submodule update
+   make archive
+   #+end_src
 
-3. Copy source tarball to =~/rpmbuild/SOURCES=.
+5. Copy source tarball to =~/rpmbuild/SOURCES=.
+   #+begin_src sh
+   mkdir -p ~/rpmbuild/SOURCES/
+   cp ./xdp-tools-$VERSION.tar.gz ~/rpmbuild/SOURCES/
+   #+end_src
 
-4. Build local package using =rpmbuild -ba packaging/rpm/xdp-tools.spec=.
+6. Build local package using =rpmbuild -ba packaging/rpm/xdp-tools.spec=.
 
-5. Check that building a scratch build on Fedora infrastructure works:
-   =fedpkg scratch-build --srpm ~/rpmbuild/SRPMS/xdp-tools-$VERSION.fcXX.src.rpm=
+7. Check that building a scratch build on Fedora infrastructure works:
+   =cd $FEDORA_DISTGIT_DIR && fedpkg scratch-build --srpm ~/rpmbuild/SRPMS/xdp-tools-$VERSION.fcXX.src.rpm=
 
 6. Sync the xdp-tools.spec file to dist-git (but don't commit anything yet):
```

Comment:
Numbering is off here now, it looks like?

Reply:
Fixed, and pushed an update!